src/spaceObjects/spaceshipParts/weaponTube.cpp

```diff
@@ -306,63 +306,108 @@ string WeaponTube::getTubeName()
     return "?" + string(direction);
 }
 
+// Coordinate transform from global reference frame (X: right/east, Y: down/south)
+// to weapon tube reference frame (X: right of tube, Y: direction of fire)
+static glm::vec2 transform_global_to_tube(glm::vec2 vec, float angle)
```

Comment:
https://github.com/daid/SeriousProton/blob/master/src/vectorUtils.h#L20

Reply:
I had considered using rotateVec2, and I would normally go with reusing a function as the preferable solution. But, I decided against it for the following reasons:
- rotateVec2 takes angles in degrees. But, there are various other operations inside the iterative loop where it would be much more convenient to keep the angles in radians by default. Thus, I thought it would be clearer if I consistently used radians inside the main body of the iterative loop, and only converted back into degrees at the very end.
- The game seems to work in a left-handed coordinate system, which I was struggling with during testing. So I wanted to switch to a right-handed system which I am more familiar with, for only the internal calculations of the function. This means a rotation + reflection transform. I thought it would be simpler to implement it directly instead of chaining rotateVec2 with a reflection operation.